auth/auth.go

```diff
@@ -60,14 +60,18 @@ type Verifier interface {
 	Verify(message []byte, signature []byte) error
 }
 
+type DelegateIdentity interface {
```

Comment:
I would change this to `Delegate`, move `Verifier()` to it, make a new interface called `Expires` with `Expired() bool` on it, then make `Identity` embed `Delegate`, `Expires` and `jwt.Claims`